test/jasmine/tests/config_test.js

```diff
@@ -546,10 +548,27 @@ describe('config argument', function() {
 
         afterEach(function() {
             destroyGraphDiv();
-            // Reset window size
             viewport.reset();
         });
 
+        function testResponsive(promise) {
+            return promise
+            .then(function() {
+                checkLayoutSize(startWidth, startHeight);
+            })
+            // Resize viewport
+            .then(function() {
+                viewport.set(newWidth, newHeight);
+            })
+            // Wait for resize to happen (Plotly.resize has a 100ms timeout)
+            .then(delay(200))
+            // Check final figure size
+            .then(function() {
+                checkLayoutSize(newWidth, newHeight);
+            })
+            .catch(failTest);
+        }
```

Comment:
Good idea to pull this function out! Nice and 🌴. You can probably simplify it a bit though:
```js
function testResponsive() {
    checkLayoutSize(startWidth, startHeight);
    viewport.set(newWidth, newHeight);

    return delay(200)()
    .then(function() {
        checkLayoutSize(newWidth, newHeight);
    })
    .catch(failTest);
}
```
And use it like:
```js
Plotly.plot(gd, data, {}, {responsive: true})
.then(testResponsive)
.then(done);
```

Reply:
Good call :) Thank you again